monday.com Raycast extension

freak4pc opened the pull request:
## Description

This is a new [monday.com](https://monday.com) Raycast extension I've been working on. 

It has a bunch of nice and productivity-enhancing features for monday.com users, the notable ones are:

1. See recently interacted-with boards, and let you directly navigate to a board
2. Quick-add an item to one of the recent boards
3. See all your team members and their contact information, quickly letting you copy or interact with

## Screencast

Recent boards flow:
![recent-add](https://user-images.githubusercontent.com/605076/155795616-f9f84d4c-c042-45d1-983b-9627ece420f4.gif)

Team flow:

![team](https://user-images.githubusercontent.com/605076/155795844-69672c50-096c-41e6-a7d0-e37deb86bcde.gif)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

pernielsentikaer commented on extensions/monday/package.json:
```diff
@@ -0,0 +1,67 @@
+    {
+      "name": "index",
+      "title": "Recent Boards",
+      "subtitle": "monday.com",
```
```suggestion
      "subtitle": "monday.",
```

freak4pc replied:
The brand name is monday.com (you can see the website), prefer leaving it as-is :)

pernielsentikaer commented on extensions/monday/package.json:
```diff
@@ -0,0 +1,67 @@
+    {
+      "name": "team",
+      "title": "My Team",
+      "subtitle": "monday.com",
```
```suggestion
      "subtitle": "monday",
```

freak4pc replied:
ditto

pernielsentikaer commented on extensions/monday/package.json:
```diff
@@ -0,0 +1,67 @@
+    },
+    {
+      "name": "open",
+      "title": "Go to my monday.com account",
+      "subtitle": "monday.com",
```
```suggestion
      "title": "Open Account",
      "subtitle": "monday",
```

freak4pc replied:
Open Account might confuse someone to think we're "opening an account for them" (i.e. creating an account for them).

How do you feel about "Navigate to my monday.com account in a browser", or some other more verbose terminology?

pernielsentikaer commented on extensions/monday/src/index.tsx:
```diff
@@ -0,0 +1,155 @@
+  cacheUser,
+} from "./lib/persistence";
+import { getBoardsAndUser } from "./lib/api";
+const BoardsCacheKey = "boards-response";
```
```suggestion
```
Unused constant

freak4pc replied:
thanks!

pernielsentikaer reviewed (changes requested):
Hi @freak4pc 👋 

Awesome extension, looking forward to seeing this in the store!

I have some suggestions for you to address:

- Media used in the README should be moved to a folder called `media` instead

- When wrong credentials, then there should be a Failure Toast (dying gracefully instead)

- Recent Boards 
  -  If credentials at any point get invalidated, then it should clear the cached result and die gracefully as above (right now I can see the cached data for a half seconds before dying)

![26CPv2cX1Dl6ysuk](https://user-images.githubusercontent.com/38033107/155861079-6eddfb28-1606-435b-8e8e-e589a994e8af.png)

- My teams
  - Dying pretty hard when credentials are wrong
  
  
![2lekOxYV2w79YEMJ](https://user-images.githubusercontent.com/38033107/155861256-62b8c466-7344-4d74-ad8f-d380e2f19e4e.png)

Since this is going to change the code significantly, I'm going to do a thorough review when you're done

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

freak4pc commented:
Hey @pernielsentikaer,

I believe I've addressed your concerns. The only thing I'd love to do is being able to reset the user's preferences on a token failure, but based on the docs there's no way to do this today ... 

I've settled on a Detail-based error view based the status code of the request:
<img width="862" alt="Screen Shot 2022-02-27 at 22 08 17" src="https://user-images.githubusercontent.com/605076/155898819-55e104b4-8c27-4ec8-b782-b676ce8dbc4d.png">

pernielsentikaer commented:
Hi @freak4pc 👋 

It's getting better and better 💪

Could you add your new error screen to "Go to My Account" also? 😄

pernielsentikaer reviewed (approved):
Awesome @freak4pc 💪 

I think we're ready to get it merged, good job 💯

sxn commented:
🎉
